File: snippets/csharp/Hello.cs
namespace Snippets;

// ANCHOR: full
using Nostr.Sdk;

public class Hello
{
    public static async Task SayHello()
    {
        // ANCHOR: client
        var keys = Keys.Generate();
        var signer = NostrSigner.Keys(keys);
        var client = new Client(signer);
        // ANCHOR_END: client
        
        // ANCHOR: connect
        var relayUrl = RelayUrl.Parse("wss://relay.damus.io");
        await client.AddRelay(relayUrl);
        await client.Connect();
        // ANCHOR_END: connect
        
        // ANCHOR: publish
        var builder = EventBuilder.TextNote("Hello, rust-nostr!");
        var output = await client.SendEventBuilder(builder);
        // ANCHOR_END: publish
        
        // ANCHOR: output
        Console.WriteLine($"Event ID: {output.id.ToBech32()}");
        Console.WriteLine($"Sent to: {string.Join(", ", output.success)}");
        Console.WriteLine($"Not sent to: {string.Join(", ", output.failed)}");
        // ANCHOR_END: output
    }
}
// ANCHOR_END: full

File: snippets/python/src/hello.py
# ANCHOR: full
import asyncio
from nostr_sdk import Keys, Client, EventBuilder, NostrSigner, RelayUrl


async def hello():
    # ANCHOR: client
    keys = Keys.generate()
    signer = NostrSigner.keys(keys)
    client = Client(signer)
    # ANCHOR_END: client

    # ANCHOR: connect
    relay_url = RelayUrl.parse("wss://relay.damus.io")
    await client.add_relay(relay_url)
    await client.connect()
    # ANCHOR_END: connect

    # ANCHOR: publish
    builder = EventBuilder.text_note("Hello, rust-nostr!")
    output = await client.send_event_builder(builder)
    # ANCHOR_END: publish

    # ANCHOR: output
    print(f"Event ID: {output.id.to_bech32()}")
    print(f"Sent to: {output.success}")
    print(f"Not send to: {output.failed}")
    # ANCHOR_END: output

if __name__ == '__main__':
   asyncio.run(hello())
# ANCHOR_END: full
